config/v1/types_oauth.go

```diff
@@ -1,37 +1,478 @@
+	// Only ONE
+	// BasicAuth contains configuration optins for the BasicAuth IdP
+	// +optional
+	BasicAuth *BasicAuthPasswordIdentityProvider `json:"basicAuth,omitempty"`
+
+	// AllowAll enables the AllowAllIdentityProvider which provides identities for users
+	// authenticating using non-empty passwords.
+	// Defaults to `false`, i.e. allowAll set to off
+	// +optional
+	AllowAll bool `json:"allowAll"`
+
+	// DenyAll enables the DenyAllPasswordIdentityProvider which provides no identities for users
+	// Defaults to `false`, ie. denyAll set to off
+	// +optional
+	DenyAll bool `json:"denyAll"`
+
+	// HTPasswd
+	// +optional
+	HTPasswd *HTPasswdPasswordIdentityProvider `json:"htpasswd,omitempty"`
+
+	// LDAP enables user authentication using LDAP credentials
+	// +optional
+	LDAP *LDAPPasswordIdentityProvider `json:"ldap,omitempty"`
+
+	// Keystone enables user authentication using keystone password credentials
+	// +optional
+	Keystone *KeystonePasswordIdentityProvider `json:"keystone,omitempty"`
+
+	// RequestHeader enables user authentication using request header credentials
+	RequestHeader *RequestHeaderIdentityProvider `json:"requestHeader,omitempty"`
+
+	// GitHub enables  user authentication using GitHub credentials
+	// +optional
+	GitHub *GitHubIdentityProvider `json:"github,omitempty"`
+
+	// GitLab enables user authentication using GitLab credentials
+	// +optional
+	GitLab *GitLabIdentityProvider `json:"gitlab,omitempty"`
+
+	// Google enables user authentication using Google credentials
+	// +optional
+	Google *GoogleIdentityProvider `json:"google,omitempty"`
+
+	// OpenID enables user authentication using OpenID credentials
+	// +optional
+	OpenID *OpenIDIdentityProvider `json:"openID,omitempty"`
+}
+
+// BasicAuthPasswordIdentityProvider provides identities for users authenticating using HTTP basic auth credentials
+type BasicAuthPasswordIdentityProvider struct {
+	// RemoteConnectionInfo contains information about how to connect to the external basic auth server
+	OAuthRemoteConnectionInfo `json:",inline"`
+}
+
+// RemoteConnectionInfo holds information necessary for establishing a remote connection
+type OAuthRemoteConnectionInfo struct {
+	// URL is the remote URL to connect to
+	URL string `json:"url"`
+	// CA is a reference to a ConfigMap containing the CA for verifying TLS connections
+	CA ConfigMapReference `json:"ca"`
+	// TLSRef is the TLS client cert information to present
+	// this is anonymous so that we can inline it for serialization
+	// The secret referenced must have a secret type SecretTypeTLS="kubernetes.io/tls"
+	// Required fields:
+	// - Secret.Data["tls.key"] - TLS private key.
+	//   Secret.Data["tls.crt"] - TLS certificate.
+	TLSRef corev1.SecretReference `json:"tls"`
+}
+
+// HTPasswdPasswordIdentityProvider provides identities for users authenticating using htpasswd credentials
+type HTPasswdPasswordIdentityProvider struct {
+	// File is a reference to your htpasswd file
+	File string `json:"file"`
```

Comment:
This won't work because they need to be able to mount the file.  Can we make this a secretRef instead?